egs2/TEMPLATE/tts1/tts.sh

```diff
@@ -255,43 +255,94 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
                 ${_opts} \
                 "${data_feats}/${dset}"
 
-            pyscripts/feats/feat-to-shape.py "scp:head -n 1 ${data_feats}/${dset}/feats.scp |" - | \
-                awk '{ print $2 }' | cut -d, -f2 > ${data_feats}/${dset}/feats_dim
+            # 3. Derive the the frame length and feature dimension
+            scripts/feats/feat_to_shape.sh --nj "${_nj}" --cmd "${train_cmd}" \
+                "${data_feats}/${dset}/feats.scp" "${data_feats}/${dset}/feats_shape"
+
+            # 4. Write feats_dim
+            head -n 1 "${data_feats}/${dset}/feats_shape" | awk '{ print $2 }' \
+                | cut -d, -f2 > ${data_feats}/${dset}/feats_dim
+
+            # 5. Write feats_type
             echo "${feats_type}" > "${data_feats}/${dset}/feats_type"
         done
     fi
 fi
 
 
-token_list="data/token_list/${trans_type}/tokens.txt"
 if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
-    log "Stage 3: Generate character level token_list from ${srctexts}"
-    mkdir -p "$(dirname ${token_list})"
+    log "Stage 3: Remove short data: ${data_feats}/* -> ${data_feats}/*_fix"
+
+    for dset in "${train_set}" "${dev_set}" ${eval_sets}; do
+        # Copy data dir
+        utils/copy_data_dir.sh "${data_feats}/${dset}" "${data_feats}/${dset}_fix"
```

Comment:
using the `_fix` suffix for all data looks ugly.
How about moving the original data under the data directory (e.g.,  `${data_feats}/${dset}/org/${dset}`) and rename the fixed directory with `${data_feats}/${dset}`